### ISSUE TITLE
Testing results

### ISSUE THREAD

**Issue by luvimperfection** [dr-pato/audio_visual_speech_enhancement#23]
Hi Dr. Pato it's me again....
(I really wish that I can close this issue by my self again....)

I have issues with the testing results.
I successfully ran training/testing code with no error, but as I checked the results by listening to the wav files inside `\SAVED\enhanced`, the results were poor. It was not separated at all.

I first thought that maybe epochs are not enough or something,
(Saw that you mentioned 10 is enough, so I just set epoch as 10) 
So I trained several epochs more, but the results were still poor.

However, I found this problem:
**same file in `MIXED/TESTING_SET/s1` and `SAVED/mixed` were different.**

Mixed speech generator worked well, I literally listened to every audios inside `MIXED/TRAINING_SET`, `MIXED/VALIDATION_SET`, `MIXED/TRAINING_SET`.

But I found that several files inside `SAVED/mixed` were not mixed at all. There were several "base wav" files inside `SAVED/mixed`.

For example, `MIXED/TESTING_SET/s1/lwbf3s_with_s3_lraq8n.wav` was a mixed audio, but `SAVED/mixed/s1_lwbf3s_with_s3_lraq8n.wav` was a clear audio.

So, the problem is that "SOMEHOW", some of `SAVED/mixed` files are not mixed audio but clear audio. So I followed your testing.py file,
**And it just points that problem might reside in tfrecord file itself.**

However, I followed your `create_dataset_tfrecords.py` file line by line, printing the data path directories, but found there were no problems...
`file_base_audio` /tf/data/GRID/s3/audio/pgaq5s.wav
`file_mix_audio`/tf/data/GRID/MIXED/TEST_SET/s3/pgaq5s_with_s2_sgbp3n.wav
`file_other_audio` /tf/data/GRID/s2/audio/sgbp3n.wav
To this point, I really don't get any sense what my problem is so..
I 'm just guessing that something strange happened in TFRecord generation part?

FYI, These are the scripts I ran.
```
# Generate mixed speech
python av_speech_enhancement.py mixed_speech_generator --data_dir /tf/data/GRID --base_speaker_ids 1 2 3 --audio_dir . --dest_dir MIXED/TRAINING_SET --num_samples 3 --num_mix 4 --num_mix_speakers 1
python av_speech_enhancement.py mixed_speech_generator --data_dir /tf/data/GRID --base_speaker_ids 1 2 3 --audio_dir . --dest_dir MIXED/VALIDATION_SET --num_samples 3 --num_mix 4 --num_mix_speakers 1
python av_speech_enhancement.py mixed_speech_generator --data_dir /tf/data/GRID --base_speaker_ids 1 2 3 --audio_dir . --dest_dir MIXED/TEST_SET --num_samples 3 --num_mix 4 --num_mix_speakers 1

# Compute power-law compressed spectrograms of mixed-speech audio samples.
python av_speech_enhancement.py audio_preprocessing --data_dir /tf/data/GRID/MIXED/TRAINING_SET --speaker_ids 1 2 3 --audio_dir . --dest_dir . -ml 48000
python av_speech_enhancement.py audio_preprocessing --data_dir /tf/data/GRID/MIXED/VALIDATION_SET --speaker_ids 1 2 3 --audio_dir . --dest_dir . -ml 48000
python av_speech_enhancement.py audio_preprocessing --data_dir /tf/data/GRID/MIXED/TEST_SET --speaker_ids 1 2 3 --audio_dir . --dest_dir . -ml 48000

# Extract face landmarks from video
python av_speech_enhancement.py video_preprocessing --data_dir /tf/data/GRID --speaker_ids 1 2 3 --video_dir video --dest_dir face_landmark --shape_predictor /tf/data/GRID/shape_predictor_68_face_landmarks.dat --ext mpg

# Computing Target Binary Masks from clean audio
python av_speech_enhancement.py tbm_computation --data_dir /tf/data/GRID --speaker_ids 1 2 3 --audio_dir audio --dest_dir tbm -ml 48000

# Generate TF Records <<FIXED>>
python av_speech_enhancement.py tfrecords_generator --data_dir /tf/data/GRID --num_speakers 2 --dest_dir tfrecords --base_audio_dir audio --video_dir face_landmark --tbm_dir tbm --mix_audio_dir MIXED --norm_data_dir /tf/data/GRID/NORM --mode fixed

# TRAIN <<FIXED>>
python av_speech_enhancement.py training --data_dir /tf/data/GRID --train_set tfrecords/TRAINING_SET --val_set tfrecords/VALIDATION_SET --exp 2 --mode fixed --num_audio_samples 48000 --model vl2m --opt adam --learning_rate 0.005 --batch_size 8 --epochs 10 -nl 1 -nh 1

# TEST <<FIXED>>
python av_speech_enhancement.py testing --data_dir /tf/data/GRID --test_set tfrecords/TEST_SET --exp 3 --num_audio_samples 48000 --mode fixed --output_dir /tf/data/GRID/eval-3 --mask_dir estimated_masks --ckp 10_44
```

Any device would be helpful

**Comment by luvimperfection** [comment]
By the way
I checked base audio, mixed audio (base audio were clean, mixed audio were mixed)
mixed audio spectrogram, target binary masks (they were valid spectrogram images)

and I even checked inside tfrecords
```
#print(example.features.feature['base_audio_path'])
bytes_list { 
  value: "/s2/audio/lbad7p.wav"
} 

#print(example.features.feature['mix_audio_path'])
bytes_list {                                                           
  value: "/MIXED/TRAINING_SET/s2/lbad7p_with_s3_pgwy6n.wav" 
}       
              
#print(example.features.feature['other_audio_path']                                           
bytes_list {                                                                      
  value: "/s3/audio/pgwy6n.wav"                   
}                     
```                           
I really don't understand why.. the testing result files are so mixed up,
base audio inside mixed dir, mixed audio inside target dir..

**Comment by dr-pato** [comment]
Hi @luvimperfection,
your problem looks very strange. I suggest you to repeat the entire pipeline and check if you get the same problem again.
There is certainly a problem in your pipeline that switches clean and mixed audios in output folders.

Best,
Giovanni

**Comment by luvimperfection** [comment]
Can you provide sample scripts you ran & your directory structure tree?
I already tried data preprocessing from scratch several times, but still cannot resolve this problem..

**Comment by dr-pato** [comment]
Hi,
I am sorry for the delay. Did you fix the problem?

**Comment by luvimperfection** [comment]
No.. I was just repeating the same process
from re-installing the environment inside docker, downloading data, preprocessing etc.
is there anything wrong with my script? (esp. with regard to tfrecords..?)
